File: problems/subsets-ii/backtracking.cs
public class Solution {
    public IList<IList<int>> SubsetsWithDup(int[] nums) {
        var subset = new List<int>();
        var results = new List<IList<int>>();

        Backtrack(0, nums, subset, results);

        return results;
    }

    private void Backtrack(int start, int[] nums, List<int> subset, List<IList<int>> results)
    {
        results.Add(new List<int>(subset));

        for (var i = start; i < nums.Length; i++)
        {
            if (start != i && nums[i] == nums[i-1]) continue;
            var num = nums[i];

            subset.Add(num);

            Backtrack(i + 1, nums, subset, results);  

            subset.RemoveAt(subset.Count - 1);
        }
    }
}

File: problems/subsets-ii/backtracking.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        sortedNums = sorted(nums)
        results = []
        subset = []
        uniqueAnswers = set()

        def backtrack(start: int) -> None:
            ans = subset[:]
            uniqueAns = ",".join(str(x) for x in ans)
            if uniqueAns not in uniqueAnswers:
                uniqueAnswers.add(uniqueAns)
                results.append(ans)

            for i in range(start, len(sortedNums)):
                num = sortedNums[i]
                subset.append(num)

                backtrack(i + 1)

                subset.pop()

        backtrack(0)

        return results
